src/inclusive_map.rs

```diff
@@ -427,6 +439,28 @@ where
         }
     }
 
+    /// Gets an iterator over all pairs of key range and value, where the key range overlaps with
+    /// the provided range.
+    ///
+    /// The iterator element type is `(&'a RangeInclusive<K>, &'a V)`.
+    pub fn range<'a>(&'a self, range: &'a RangeInclusive<K>) -> RangeIter<'a, K, V> {
+        use core::ops::Bound;
+
+        let start = self
+            .get_key_value(range.start())
+            .map_or(range.start(), |(k, _v)| k.start());
+        let end = range.end();
+
+        RangeIter {
+            inner: self.btm.range((
+                Bound::Included(RangeInclusiveStartWrapper::new(
```

Comment:
This is the bit that looks suspicious to me: what happens if a stored range starts before the start of the given `range`? (I'll add a test to cover this after sprucing up the existing test patterns a bit.)